proposals/csharp-10.0/parameterless-struct-constructors.md

```diff
@@ -0,0 +1,102 @@
+# Parameterless struct constructors
+
+## Summary
+
+Support explicit parameterless constructors for value types.
+
+## Proposal
+
+If the parameterless constructor is less accessible than the value type, a warning is reported and the constructor is ignored when inaccessible.
+```csharp
+struct NoConstructor
+{
+    // ok
+}
+struct PublicConstructor
+{
+    internal PublicConstructor() { } // ok: always accessible
+}
+struct PrivateConstructor
+{
+    private PrivateConstructor() { } // warning: parameterless .ctor will be ignored if inaccessible
+}
+```
+
+`default(S)` ignores any parameterless constructor and generates a zeroed `S`.
+```csharp
+NoConstructor x = default;      // ok
+PublicConstructor y = default;  // ok
+PrivateConstructor z = default; // ok
+```
+
+For a type parameter `T` with a `new()` constraint, `new T()` is emitted as a call to `System.Activator.CreateInstance<T>()` which ignores any explicit parameterless constructor.
```

Comment:
My understanding is that this bug in the CLR was fixed:

https://github.com/dotnet/coreclr/pull/10778

Reply:
Also found this from a recent LDM:

https://github.com/dotnet/csharplang/blob/master/meetings/2021/LDM-2021-01-27.md#field-initializers

> We're still interested in doing this: the `Activator.CreateInstance` bug was in a version of the framework that is long out of support at this point, and we have universal agreement behind the idea.

Reply:
Think we should have an open issue section at the bottom to follow up with CLR on this point.